src/main/java/com/datalogics/pdf/samples/manipulation/ConvertToPdfA2.java

```diff
@@ -0,0 +1,244 @@
+            if ("a".equals(conformanceLevelInput)) {
+                conformanceLevel = PDFA2ConformanceLevel.Level_2a;
+            } else if ("b".equals(conformanceLevelInput)) {
+                // this isn't necessary since it defaults to 2b, for clarity though it is here
+                conformanceLevel = PDFA2ConformanceLevel.Level_2b;
+            } else if ("u".equals(conformanceLevelInput)) {
+                conformanceLevel = PDFA2ConformanceLevel.Level_2u;
+            } else {
+                throw new IllegalArgumentException("Valid 'a', 'b', or 'u' PDF/A2 conformance level not specified.");
+            }
+        } else {
+            inputUrl = ConvertPdfDocument.class.getResource(INPUT_PDF_PATH);
+            outputUrl = IoUtils.createUrlFromPath(OUTPUT_PDF_PATH);
+        }
+
+        convertToPdfA2(inputUrl, outputUrl, conformanceLevel);
+        validate(outputUrl, conformanceLevel);
+    }
+
+    /**
+     * Converts an input PDF document to the PDF/A-2 standard with the specified conformance level.
+     *
+     * @param inputUrl The URL of the document to be converted
+     * @param outputUrl The URL of the converted document
+     * @throws IOException an I/O operation failed or was interrupted
+     * @throws PDFFontException there was an error in the font set or an individual font
+     * @throws PDFInvalidDocumentException a general problem with the PDF document, which may now be in an invalid state
+     * @throws PDFIOException there was an error reading or writing a PDF file or temporary caches
+     * @throws PDFSecurityException some general security issue occurred during the processing of the request
+     * @throws PDFInvalidParameterException one or more of the parameters passed to a method is invalid
+     * @throws PDFUnableToCompleteOperationException the operation was unable to be completed
+     */
+    public static void convertToPdfA2(final URL inputUrl, final URL outputUrl,
+                                      final PDFA2ConformanceLevel conformanceLevel)
+                    throws IOException, PDFFontException, PDFInvalidDocumentException, PDFIOException,
+                    PDFSecurityException, PDFInvalidParameterException, PDFUnableToCompleteOperationException {
+        ByteWriter writer = null;
+        // Attach font set to PDF
+        final PDFFontSet pdfaFontSet = FontSetLoader.newInstance().getFontSet();
+        final PDFOpenOptions openOptions = PDFOpenOptions.newInstance();
+        openOptions.setFontSet(pdfaFontSet);
+
+        final PDFDocument pdfDoc = DocumentUtils.openPdfDocumentWithOptions(inputUrl, openOptions);
+
+        final PDFA2ConversionOptions options = new PDFA2ConversionOptions();
+        final PdfA2ConversionHandler handler = new PdfA2ConversionHandler();
+
+        // Tell font set to embed fonts
+        options.setShouldEmbedFonts(true);
+
+        // Set an OutputIntent
+        if (options.getPDFAOutputIntent() == null) {
```

Comment:
@This whole section should probably be added to PDFJT itself, where it can use the built-in color profiles.

It's very similar to code in the ``PDFAConversionOptionsFactory`` (see [here](https://octocat.dlogics.com/datalogics/pdf-java-toolkit/blob/develop/pdfjt/src/main/java/com/adobe/pdfjt/services/pdfa/PDFAConversionOptionsFactory.java#L113)), which makes me think what we really need is a similar factory for PDF/A2.